### PR TITLE
[yaml] #1542 Implement DrupalStyle

### PR DESCRIPTION
**Pull request by mnico**

### CODE REVIEW

File: src/Command/Yaml/DiffCommand.php [hunk 1]
```diff
@@ -177,35 +170,52 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-
         // --yaml-left option
         $yaml_left = $input->getArgument('yaml-left');
         if (!$yaml_left) {
-            $yaml_left = $output->ask(
-                $this->trans('commands.yaml.diff.questions.yaml-left'),
-                null,
-                $validator_filename
-            );
+            while (true) {
```
Comment (enzolutions) [inline src/Command/Yaml/DiffCommand.php]: Hi @mnico 

I don't understand why we need a while(true) if the ask fail the cycle never ends

File: src/Command/Yaml/DiffCommand.php [hunk 2]
```diff
@@ -177,35 +170,52 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
 
         // --yaml-right option
         $yaml_right = $input->getArgument('yaml-right');
         if (!$yaml_right) {
-            $yaml_right = $output->ask(
-                $this->trans('commands.yaml.diff.questions.yaml-right'),
-                null,
-                $validator_filename
-            );
+            while (true) {
```
Comment (enzolutions) [inline src/Command/Yaml/DiffCommand.php]: Same question above why a white(true)

File: src/Command/Yaml/MergeCommand.php [hunk 3]
```diff
@@ -132,11 +136,18 @@ protected function interact(InputInterface $input, OutputInterface $output)
         // --yaml-destination option
         $yaml_destination = $input->getArgument('yaml-destination');
         if (!$yaml_destination) {
-            $yaml_destination = $output->ask(
-                $this->trans('commands.yaml.merge.questions.yaml-destination'),
-                '',
-                $validator_filename
-            );
+            while (true) {
```
Comment (enzolutions) [inline src/Command/Yaml/MergeCommand.php]: same, why while(true)

File: src/Command/Yaml/SplitCommand.php [hunk 4]
```diff
@@ -147,24 +152,27 @@ protected function interact(InputInterface $input, OutputInterface $output)
         // --yaml-left option
         $yaml_file = $input->getArgument('yaml-file');
         if (!$yaml_file) {
-            $yaml_file = $output->ask(
-                $this->trans('commands.yaml.diff.questions.yaml-left'),
-                '',
-                $validator_filename
-            );
+            while (true) {
```
Comment (enzolutions) [inline src/Command/Yaml/SplitCommand.php]: while?

File: src/Command/Yaml/DiffCommand.php [hunk 5]
```diff
@@ -177,35 +170,52 @@ protected function execute(InputInterface $input, OutputInterface $output)
      */
     protected function interact(InputInterface $input, OutputInterface $output)
     {
-        $validator_filename = function ($value) {
+        $io = new DrupalStyle($input, $output);
+
+        $validator_filename = function ($value) use ($io) {
             if (!strlen(trim($value))) {
-                throw new \Exception(' You must provide a valid file path.');
```
Comment (mnico) [inline src/Command/Yaml/DiffCommand.php]: @enzolutions Hi, The use of "while" is added because I removed the throw exception in the validator

Comment (mnico) [inline src/Command/Yaml/DiffCommand.php]: And with the "while" if the user enters an invalid path then the system will ask again for a valid path.

**Comment by enzolutions** [comment]
Thanks @mnico for your contribution

**Pull request merged**